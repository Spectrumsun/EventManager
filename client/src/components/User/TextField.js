import React from 'react';

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
      //required
    />
  </div>
);


export default TextField;
