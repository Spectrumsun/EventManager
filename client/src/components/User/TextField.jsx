import React from 'react';
import PropTypes from 'prop-types'; 

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
    
    />
  </div>
);


TextField.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default TextField;
