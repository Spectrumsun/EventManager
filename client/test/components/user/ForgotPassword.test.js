import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import ConnectedForgotPassword,
{ ForgotPassword } from '../../../src/components/User/ForgotPassword';
import TextField from '../../../src/components/User/TextField'

const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  isAuthenticated: false,
  user: {},
};
const store = mockStore(initialState);

const props = {
  initconfirmPassword: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory(),
  type: 'PropTypes.string.isRequired',
  value:' PropTypes.string.isRequired',
  onChange: () => {},
  name: 'PropTypes.string.isRequired',
  label: 'PropTypes.string.isRequired',
  placeholder: 'PropTypes.string.isRequired'
};

const mountedWrapper = mount(
  <Provider store={store}>
    <BrowserRouter>
      <ConnectedForgotPassword {...props} />
    </BrowserRouter>
  </Provider>
);

const shallowWrapper = shallow(<ForgotPassword {...props} />);

const state = {
  email: 'bot@yahoo.com',
};

const event = {
  preventDefault: jest.fn(),
  target: {
    email: 'bot@yahoo.com',
  }
};

describe('<ForgotPassword /> Component', () => {
  it('should render the <Login />', () => {
    shallow(<ForgotPassword />);
  });

  it('should render the <ForgotPassword /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('ForgotPassword').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <ForgotPassword {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });

  it('should render initial layout of ForgotPassword', () => {
    const wrapper = shallow(<ForgotPassword />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have three image on layout', () => {
    const wrapper = shallow(<ForgotPassword />);
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should have <TextField /> when the page loads', () => {
    const wrapper = shallow(<ForgotPassword />);
    expect(wrapper.find(TextField)).toHaveLength(1);
  });

  it('calls onChange event when input is passed to state', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event when submit button is clicked', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });
});
