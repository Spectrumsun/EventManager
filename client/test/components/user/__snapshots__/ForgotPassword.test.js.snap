// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> Component should match component snapshot 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "paddingTop": "150px",
      }
    }
  >
    <h3
      className="center"
    >
      Event Manager
    </h3>
    <div
      className="card loginCard"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <div
        className="card-header"
      >
        <h3>
          Password Reset
        </h3>
      </div>
      <div
        className="card-body"
      >
        <div
          className="cont card-body"
        >
          <form
            className="centerform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Email address
              </label>
              <input
                className="form-control form-control-lg"
                name="email"
                onChange={[Function]}
                placeholder="Vaild Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="text-center"
            >
              <button
                className="btn btn-outline-dark"
                type="submit"
              >
                Email me a recovery link
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ForgotPassword /> Component should render initial layout of ForgotPassword 1`] = `
Array [
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "150px",
        }
      }
    >
      <h3
        className="center"
      >
        Event Manager
      </h3>
      <div
        className="card loginCard"
        style={
          Object {
            "width": "30rem",
          }
        }
      >
        <div
          className="card-header"
        >
          <h3>
            Password Reset
          </h3>
        </div>
        <div
          className="card-body"
        >
          <div
            className="cont card-body"
          >
            <form
              className="centerform"
              onSubmit={[Function]}
            >
              <TextField
                label="Email address"
                name="email"
                onChange={[Function]}
                placeholder="Vaild Email"
                type="email"
                value=""
              />
              <div
                className="text-center"
              >
                <button
                  className="btn btn-outline-dark"
                  type="submit"
                >
                  Email me a recovery link
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
